python/src/aiconfig/schema.py

```diff
@@ -444,30 +445,134 @@ def get_model_metadata(
     # `update_model_parser_registry_with_config_runtime`` function
     # Tracked in https://github.com/lastmile-ai/aiconfig/issues/503
     def update_model(
-        self, model_metadata: Dict | ModelMetadata, prompt_name: Optional[str] = None
+        self, name: Optional[str] = None, settings: Optional[InferenceSettings] = None, prompt_name: Optional[str] = None
```

Comment:
nit: change `name` to `model_name` or a similar for clarity, The attribute `name` is not self-explanatory without consulting the docstring.